### PR TITLE
feat: fork session support

### PR DESCRIPTION
**Pull request by darkskygit**

### CODE REVIEW

**Comment by darkskygit** [comment]
* **#7367** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7367?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @darkskygit and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/server/src/plugins/copilot/resolver.ts [hunk 1]
```diff
@@ -313,6 +335,34 @@ export class CopilotResolver {
...
+  ) {
+    await this.permissions.checkCloudPagePermission(
+      options.workspaceId,
+      options.docId,
+      user.id
+    );
+    const lockFlag = `${COPILOT_LOCKER}:session:${user.id}:${options.workspaceId}`;
+    await using lock = await this.mutex.lock(lockFlag);
+    if (!lock) {
+      return new TooManyRequestsException('Server is busy');
```
Comment (forehalo) [inline packages/backend/server/src/plugins/copilot/resolver.ts]: do not throw http exception anymore

File: packages/backend/server/schema.prisma [hunk 2]
```diff
@@ -481,15 +481,17 @@ model AiSessionMessage {
...
-  tokenCost   Int       @default(0)
-  createdAt   DateTime  @default(now()) @map("created_at") @db.Timestamptz(6)
-  deletedAt   DateTime? @map("deleted_at") @db.Timestamptz(6)
+  id              String    @id @default(uuid()) @db.VarChar(36)
+  userId          String    @map("user_id") @db.VarChar(36)
+  workspaceId     String    @map("workspace_id") @db.VarChar(36)
+  docId           String    @map("doc_id") @db.VarChar(36)
+  promptName      String    @map("prompt_name") @db.VarChar(32)
+  // the session id of the parent session if this session is a forked session
+  parentSessionId String?   @map("parent_session_id") @db.VarChar(36)
```
Comment (forehalo) [inline packages/backend/server/schema.prisma:490]: reason?

Comment (darkskygit) [reply packages/backend/server/schema.prisma:490]: for each doc, we will only have one session for the side panel, we can call it "main session"
all sessions used in the chat block are forked from other forked sessions or the main session
this field used to identify the session is fork from which session, so we can know if the session is main session or fork from other session

**Review by forehalo — APPROVED** [review]

**Pull request merged**